src/components/Inventory/view/BatchDetailsModal.tsx

```diff
@@ -54,6 +63,30 @@ export default function BatchDetailsModal(props: BatchDetailsModalProps): JSX.El
     }
   }, [record, selectedOrganization, speciesId]);
 
+  useEffect(() => {
+    if (record?.facilityId) {
+      const newFacility = getNurseryById(selectedOrganization, record.facilityId);
+      if (newFacility.id.toString() !== facility?.id.toString()) {
+        setFacility(newFacility);
+      }
+    }
+  }, [record?.facilityId, selectedOrganization, facility?.id]);
+
+  useEffect(() => {
+    if (timeZone !== tz.id) {
+      setTimeZone(tz.id);
+    }
+  }, [tz.id, timeZone]);
+
+  useEffect(() => {
+    setRecord((previousRecord: CreateBatchRequestPayload): CreateBatchRequestPayload => {
+      return {
+        ...previousRecord,
+        addedDate: getTodaysDateFormatted(timeZone),
```

Comment:
we should be fetching the date from the record yes?

Reply:
Yes, the only problem of doing that is that if you have the default today's date selected and then you change to another nursery that has other time zone, the date will remain with the old "today's date". That's why I added this here. I can remove it, and it will keep the record value

Reply:
If user edited the value, we should leave it as-is, otherwise make this change. We could maybe use a boolean state to track it?